sklearn/discriminant_analysis.py

```diff
@@ -120,11 +120,14 @@ def _class_cov(X, y, priors=None, shrinkage=None):
         Class covariance matrix.
     """
     classes = np.unique(y)
-    covs = []
-    for group in classes:
+    cov = np.zeros((X.shape[1], X.shape[1]), dtype=X.dtype)
+    for idx, group in enumerate(classes):
         Xg = X[y == group, :]
-        covs.append(np.atleast_2d(_cov(Xg, shrinkage)))
-    return np.average(covs, axis=0, weights=priors)
+        if priors is None:
+            cov = cov + np.atleast_2d(_cov(Xg, shrinkage)) / len(classes)
+        else:
+            cov = cov + priors[idx] * np.atleast_2d(_cov(Xg, shrinkage))
```

Comment:
I'm not familiar with this part of the code base, but if I understand correctly #10898 was only about memory usage and I don't see how this line can be equivalent to the one above (if priors=None).